src/Setup/WixBA/RootView.xaml.cs

```diff
@@ -35,7 +36,37 @@ public RootView(RootViewModel viewModel)
         /// <param name="e"></param>
         private void Background_MouseLeftButtonDown(object sender, MouseButtonEventArgs e)
         {
-            this.DragMove();
+            if (MouseButtonState.Pressed == Mouse.LeftButton)
+            {
+                this.DragMove();
+                e.Handled = true;
+            }
+        }
+ 
+        /// <summary>
+        /// If user clicks on a no-input control (i.e. a TextBlock, etc.) route event to here to avoid  InvalidOperationException
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void RootView_IgnoreClick(object sender, MouseButtonEventArgs e)
+        {
+            //e.Handled = true;
+        }
+
+        /// <summary>
+        /// Event is fired when the window is closing.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Window_Closing(object sender, System.ComponentModel.CancelEventArgs e)
+        {
+            RootViewModel rvm = this.DataContext as RootViewModel;
+            if ((null != rvm) && (InstallationState.Applying == rvm.InstallState))
+            {
+                rvm.Canceled = true;
```

Comment:
I can't remember if WixBA prompts when the user clicks the Cancel button.  If so, we also should prompt here.  If not, there's probably a bug somewhere asking us to prompt.